utils/appliance/__init__.py

```diff
@@ -483,6 +497,21 @@ def ssh_client(self):
             'container': self.container,
         }
         ssh_client = ssh.SSHClient(**connect_kwargs)
+        try:
+            ssh_client.get_transport().is_active()
+            logger.error('default appliance ssh credentials are valid')
+        except Exception as e:
+            logger.error(e)
+            logger.error('default appliance ssh credentials failed, trying establish ssh connection'
+                         ' using ssh private key')
+            with open(conf_path.join('appliance_private_key').strpath, 'w') as key:
+                key.write(conf.credentials['ssh']['private_key'])
+            connect_kwargs = {
+                'hostname': self.hostname,
+                'username': conf.credentials['ssh']['ssh-user'],
+                'key_filename': conf_path.join('appliance_private_key').strpath,
+            }
+            ssh_client = ssh.SSHClient(**connect_kwargs)
```

Comment:
@mshriver Agree :+1: I can port that to new method and handle any new exceptions there.